services/ec2/service.go

```diff
@@ -6,58 +6,70 @@ import (
 	"sync"
 	"time"
 
+	"github.com/influxdata/kapacitor/services/ec2/client"
 	"github.com/influxdata/kapacitor/services/scraper"
+	plog "github.com/prometheus/common/log"
 	"github.com/prometheus/prometheus/config"
 	pec2 "github.com/prometheus/prometheus/discovery/ec2"
 )
 
-type Diagnostic scraper.Diagnostic
+// Doesn't actually get used, but its good to have a template here already
+type Diagnostic interface {
+	WithClusterContext(cluster string) Diagnostic
+	//Debug(...interface{})
+	plog.Logger
+}
```

Comment:
This needs to stay a scraper.Diagnostic.  You can add methods to it like this:

```
type Diagnostic interface {
     scraper.Diagnostic
     WithRegionContext(region string) Diagnostic
}
```

Note I change the name to region since that what each EC2 configuration represents is a region.